src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,300 @@
+package duke.parser;
+
+import duke.ui.UserInterface;
+import duke.task.TaskList;
+import duke.exception.DukeException;
+
+import java.time.LocalDate;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is responsible for the parsing of all user inputs, as well as file
+ * contents back into tasks during the initial loading of file.
+ */
+public class Parser {
+
+    private static final String[] months = new String[] { "JANUARY", "FEBRUARY", "MARCH",
+        "APRIL", "MAY", "JUNE", "JULY", "AUGUST", "SEPTEMBER", "OCTOBER", "NOVEMBER",
+        "DECEMBER" };
+    private static final int OFFSET = 7;
+
+    /**
+     * A default constructor for the Parser class.
+     */
+    public Parser() {}
+
+    private final static UserInterface USER_INTERFACE = new UserInterface();
+
+    /**
+     * This method is responsible for detecting what the user wants. It allows the
+     * program to respond based on the first input word by the user.
+     *
+     * @param command the input provided by the user through the command line
+     * @param scanner the scanner that reads in the user's input
+     * @return true if there is a request to terminate execution and false otherwise
+     */
+    public boolean firstCommandParser(String command, Scanner scanner) {
+        TaskList task = new TaskList();
+
+        switch (command) {
+        case "bye":
+            USER_INTERFACE.exit();
+            return true;
+        case "done":
+            String stringIndex = scanner.next();
+            try {
+                int index = Integer.parseInt(stringIndex) - 1;
+                if (task.markDone(index)) {
+                    USER_INTERFACE.taskComplete(TaskList.getTask(index));
+                }
+                break;
+            } catch (NumberFormatException e) {
+                USER_INTERFACE.integerInputWarning();
+                break;
+            }
+        case "list":
+            TaskList.displayList();
+            break;
+        case "delete":
+            try {
+                String stringIndex2 = scanner.next();
+                int index2 = Integer.parseInt(stringIndex2) - 1;
+                TaskList deletedTask = TaskList.getTask(index2);
+                if (deletedTask == null) {
+                    break;
+                }
+                task.delete(index2);
+                USER_INTERFACE.taskDeleted(deletedTask);
+                break;
+            } catch (NumberFormatException e) {
+                USER_INTERFACE.integerInputWarning();
+                break;
+            }
```

Comment:
Perhaps you can consider extracting the `break` instead of repeating it in both `try` and `catch` block?